Library/Homebrew/utils/github.rb

```diff
@@ -292,7 +273,35 @@ def print_pull_requests_matching(query)
   end
 
   def private_repo?(full_name)
-    uri = URI.parse("#{API_URL}/repos/#{full_name}")
+    uri = url_to "repos", full_name
     open(uri) { |json| json["private"] }
   end
+
+  def query_string(*main_params, **qualifiers)
+    params_list = main_params
+
+    qualifiers.each do |key, value|
+      if value.is_a? Array
+        value.each { |v| params_list << format_parameter(key, v) }
+      else
+        params_list << format_parameter(key, value)
+      end
+    end
+
+    "q=#{uri_escape(params_list.join(" "))}&per_page=100"
+  end
+
+  def format_parameter(key, value)
+    "#{key}:#{value}"
+  end
+
+  def url_to(*subroutes)
+    URI.parse(File.join(API_URL, *subroutes))
```

Comment:
Agreed that this is kinda weird -- I did it so that `url_to '/repos', '/Homebrew'` would work, but I'm also fine if the weird-ness outweighs that benefit